### ISSUE TITLE
branch create incorrectly escapes branch name when writing to credentials.json

### ISSUE THREAD

**Issue by aljazerzen** [edgedb/edgedb-cli#1226]
Tracked by #1204 

If you use CLI with an pre-5.0 server, branch commands should either:
- error out saying "this instance does not support branches. please upgrade the instance to version >=5.0"
- use database commands instead, i.e. `edgedb branch create hello --empty` -> `CREATE DATABASE hello;`

**Comment by aljazerzen** [comment]
I was wrong: this is not caused by the branch already existing, but something else.

The error for branch existing is this:
```
➜  demo10 git:(feat-links) ✗ edgedb branch create feature-branch
Creating branch 'feature-branch'...
OK: CREATE BRANCH
➜  demo10 git:(feat-links) ✗ edgedb branch create feature-branch
Creating branch 'feature-branch'...
edgedb error: DuplicateDatabaseDefinitionError: branch "feature-branch" already exists
  Server traceback:
      edb.errors.DuplicateDatabaseDefinitionError: branch "feature-branch" already exists
```
... which is fine.


The problem that I hit was that escaped branch name `` `feat-links` `` was written into `credentials.json` instead of just `feat-links`.

**Comment by aljazerzen** [comment]
I was wrong: this is not caused by the branch already existing, but something else.

The error for branch existing is this:
```
➜  demo10 git:(feat-links) ✗ edgedb branch create feature-branch
Creating branch 'feature-branch'...
OK: CREATE BRANCH
➜  demo10 git:(feat-links) ✗ edgedb branch create feature-branch
Creating branch 'feature-branch'...
edgedb error: DuplicateDatabaseDefinitionError: branch "feature-branch" already exists
  Server traceback:
      edb.errors.DuplicateDatabaseDefinitionError: branch "feature-branch" already exists
```
... which is fine.


The problem that I hit was that escaped branch name `` `feat-links` `` was written into `credentials.json` instead of just `feat-links`.

**Comment by aljazerzen** [comment]
This issue was actually closed by #1234

(Yesterday I've filed a few issues, then realized that I one of them is actually some other bug, and then edited the wrong issue! This created a whole schurfuffel of confusion, but I think I restored everything now.)

We might also want to implement the second point from this issue though, as a quality of life improvement.
